test/functorch/test_aotdispatch.py

```diff
@@ -704,6 +704,32 @@ def f(a):
         self.assertEqual(x_ref.grad, x_test.grad)
         self.assertEqual(x_ref_view.grad, x_test_view.grad)
 
+    def test_nested_subclasses(self):
```

Comment:
I think the E2E test we should try before landing is probably seeing if this PR fixes the Float8 + FSDP issue here: https://github.com/pytorch-labs/float8_experimental/pull/266#issue-2309035931

I was able to repro it by cloning the `float8_experimental` repro, checking out that PR, and running the test command in the summary. Let me know if you run into issues!